### PR TITLE
status overview response data fields typing 

### PR DESCRIPTION
**Pull request by sergej-tschak**

### CODE REVIEW

File: src/userStatusOverview/types.ts [hunk 1]
```diff
@@ -1,26 +1,36 @@
...
   /**
    * Time tracking id of a running task
    */
-  time_tracking_id?: number;
-  time_tracking_start_time?: string;
-  time_tracking_start_time_timezone_id?: string;
-  time_tracking_task_id?: number;
-  time_tracking_is_nonworking?: boolean;
+  time_tracking_id: number;
+  time_tracking_start_time: string | null;
```
Comment (benj56) [inline src/userStatusOverview/types.ts:15]: I'd prefer if we could use optional instead of null, but I'm not sure how we could do that. eg. if we should change null to undefined in responseHandler

File: src/deltaSync/types.ts [hunk 2]
```diff
@@ -2,12 +2,14 @@ import { User } from '../users/types';
...
 import { Project } from '../projects/types';
 import { TimeTracking } from '../timetrackings/types';
+import { UserStatusOverview } from '../userStatusOverview/types';
 
 export type SyncData = {
   users: User[];
   tasks: Task[];
   projects: Project[];
   timeTrackings: TimeTracking[];
+  userStatusOverviews: UserStatusOverview[];
```
Comment (haubitzer) [inline src/deltaSync/types.ts:12]: Why is this needed?

File: src/index.ts [hunk 3]
```diff
@@ -31,7 +31,7 @@ export { Team } from './teams/types';
 export { TimesheetAccounting } from './timesheetAccountings/types';
 export { TimeTracking, TimeTrackingCreate, StartTimeTrackingData, StopTimeTrackingData } from './timetrackings/types';
 export { User } from './users/types';
-export { UserStatusOverview } from './userStatusOverview/types';
+export { UserStatusOverview, UserStatusOverviewStatus } from './userStatusOverview/types';
```
Comment (haubitzer) [inline src/index.ts:34]: Can we replace this with export * from ./userStatus... ?

File: src/userStatusOverview/types.ts (hunk 1 above)
Comment (haubitzer) [reply src/userStatusOverview/types.ts:15]: would also prefer optional instead of null.

**Review by haubitzer — CHANGES_REQUESTED** [review]

File: src/userStatusOverview/types.ts [hunk 4]
```diff
@@ -1,26 +1,36 @@
+export enum UserStatusOverviewStatus {
+  Offline,
```
Comment (haubitzer) [inline src/userStatusOverview/types.ts:2]: Add to all fields a value, please

**Review by haubitzer — CHANGES_REQUESTED** [review]

File: src/userStatusOverview/types.ts [hunk 5]
```diff
@@ -1,26 +1,36 @@
...
   /**
    * Current absence ids
    */
-  current_absence_ids?: string;
+  current_absence_ids: string | null;
   /**
    * Represents current status of users activity if he is offline (0), working(1),break(2),on leave(3), core time violation (5)
    */
-  status?: number;
+  status: UserStatusOverviewStatus;
```
Comment (haubitzer) [inline src/userStatusOverview/types.ts:33]: Do we know why these two fields were not part of the generated type?

Another question is why do we need it here anyway? is this information not saved in the user type?

**Review by haubitzer — CHANGES_REQUESTED** [review]

File: src/userStatusOverview/types.ts (hunk 4 above)
Comment (sergej-tschak) [reply src/userStatusOverview/types.ts:2]: this kind of declaration is allowed by typescript. typescript do auto increment of value itself

File: src/deltaSync/types.ts (hunk 2 above)
Comment (sergej-tschak) [reply src/deltaSync/types.ts:12]: will recheck this one

File: src/userStatusOverview/types.ts (hunk 1 above)
Comment (sergej-tschak) [reply src/userStatusOverview/types.ts:15]: as I remember we talked that we will clarify types for fields, so if we switch to optional then it means that we allow field to be undefined

File: src/index.ts (hunk 3 above)
Comment (sergej-tschak) [reply src/index.ts:34]: will change this

File: src/deltaSync/types.ts (hunk 2 above)
Comment (sergej-tschak) [reply src/deltaSync/types.ts:12]: because status overview will be periodically sync anyway. if approach of syncing is changed in future then it should be removed from here, but for now I think it is ok

Comment (sergej-tschak) [reply src/deltaSync/types.ts]: removed as it was already removed from FE repo

File: src/userStatusOverview/types.ts (hunk 5 above)
Comment (benj56) [reply src/userStatusOverview/types.ts:33]: We are guessing it's not in Swagger then, but we would need to ask ScrummyBears how they generated it. 
It is, but it is in this endpoint too, maybe to avoid having to do another request on clients that might not have all users stored.

**Review by benj56 — APPROVED** [review]

**Review by AlGvrl — APPROVED** [review]

File: src/userStatusOverview/types.ts (hunk 1 above)
Comment (benj56) [reply src/userStatusOverview/types.ts:15]: We'd have to look into how to do it in a separate story

File: src/userStatusOverview/types.ts (hunk 4 above)
Comment (haubitzer) [reply src/userStatusOverview/types.ts:2]: I know that is allowed but I think that we should not rely on the order here.

File: src/userStatusOverview/types.ts (hunk 5 above)
Comment (haubitzer) [reply src/userStatusOverview/types.ts:33]: Okay...  anyway, we should not use them in the react app, because there we want to get all the usernames always from our userStore.

**Review by haubitzer — APPROVED** [review]
